datafusion/core/src/datasource/listing/table.rs

```diff
@@ -1159,11 +1159,6 @@ mod tests {
         // (file_sort_order, expected_result)
         let cases = vec![
             (vec![], Ok(vec![])),
-            // not a sort expr
-            (
-                vec![vec![col("string_col").sort(true, false)]],
-                Err("Expected Expr::Sort in output_ordering, but got string_col"),
-            ),
```

Comment:
test case removed because the API now makes it impossible to provide wrong expression type